Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/connectmgr/persistance/dao/ConnectionManagerDAOBase.java

```diff
@@ -37,41 +38,79 @@
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import javax.xml.parsers.SAXParserFactory;
+import org.xml.sax.SAXException;
+import javax.xml.transform.sax.SAXSource;
+import org.xml.sax.InputSource;
+import org.xml.sax.XMLReader;
+import java.io.Reader;
+import java.io.FileNotFoundException;
 
 /**
  *
  * @author mweaver
  */
 public class ConnectionManagerDAOBase {
 
-    private static final Logger LOG = LoggerFactory.getLogger(ConnectionManagerDAOBase.class);
-
-    protected BusinessDetail loadBusinessDetail(File file) throws JAXBException {
-        BusinessDetail resp;
-        synchronized (file) {
-            JAXBContext context = JAXBContext.newInstance(BusinessDetail.class);
-            Unmarshaller unmarshaller = context.createUnmarshaller();
-            JAXBElement<BusinessDetail> jaxbElement = unmarshaller.unmarshal(new StreamSource(file),
-                    BusinessDetail.class);
-            resp = jaxbElement.getValue();
-        }
-        return resp;
-    }
-
-    protected void saveBusinessDetail(BusinessDetail BusinessDetail, File file) {
-        try {
-            synchronized (file) {
-                JAXBContext context = JAXBContext.newInstance(BusinessDetail.class);
-                ObjectFactory factory = new ObjectFactory();
-                Marshaller marshaller = context.createMarshaller();
-                marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
-                marshaller.marshal(factory.createBusinessDetail(BusinessDetail), file);
-            }
-        } catch (JAXBException ex) {
-            throw new RuntimeException("Unable to save to Connection Information File " + file.getName(), ex);
-        }
-
-        LOG.info("Connection info saved to " + file.getName());
-    }
+	private static final Logger LOG = LoggerFactory.getLogger(ConnectionManagerDAOBase.class);
+
+	protected BusinessDetail loadBusinessDetail(File file) throws JAXBException {
+		BusinessDetail resp = null;
+		try {
+
+			synchronized (file) {
+
+				SAXParserFactory spf = SAXParserFactory.newInstance();
+				spf.setNamespaceAware(true);
+				spf.setFeature(NhincConstants.FEATURE_GENERAL_ENTITIES, false);
+				spf.setFeature(NhincConstants.FEATURE_DISALLOW_DOCTYPE, true);
+				spf.setFeature(NhincConstants.FEATURE_PARAMETER_ENTITIES, false);
+
+				XMLReader xmlReader = spf.newSAXParser().getXMLReader();
+				InputSource inputSource = new InputSource(new FileReader(file));
+				SAXSource source = new SAXSource(xmlReader, inputSource);
+
+				JAXBContext context = JAXBContext.newInstance(BusinessDetail.class);
+				Unmarshaller unmarshaller = context.createUnmarshaller();
+
+				InputSource stream = source.getInputSource();
+
+				Reader reader = stream.getCharacterStream();
+
+				JAXBElement<BusinessDetail> jaxbElement = unmarshaller.unmarshal(new StreamSource(reader),
+						BusinessDetail.class);
+
+				resp = jaxbElement.getValue();
+
+			}
+		} catch (ParserConfigurationException e) {
```

Comment:
group all exception similar like this "catch (ParserConfigurationException | SAXException | FileNotFoundException e) "